src/renderers/dom/shared/DOMProperty.js

```diff
@@ -197,33 +199,60 @@ var DOMProperty = {
   properties: {},
 
   /**
-   * Mapping from lowercase property names to the properly cased version, used
-   * to warn in the case of missing properties. Available only in __DEV__.
-   *
-   * autofocus is predefined, because adding it to the property whitelist
-   * causes unintended side effects.
-   *
-   * @type {Object}
-   */
-  getPossibleStandardName: __DEV__ ? {autofocus: 'autoFocus'} : null,
-
-  /**
-   * All of the isCustomAttribute() functions that have been injected.
+   * Some attributes are aliased for easier use within React. We don't
+   * allow direct use of these attributes. See DOMAttributeNames in
+   * HTMLPropertyConfig and SVGPropertyConfig.
    */
-  _isCustomAttributeFunctions: [],
+  aliases: {},
 
   /**
-   * Checks whether a property name is a custom attribute.
+   * Checks whether a property name is a writeable attribute.
    * @method
    */
-  isCustomAttribute: function(attributeName) {
-    for (var i = 0; i < DOMProperty._isCustomAttributeFunctions.length; i++) {
-      var isCustomAttributeFn = DOMProperty._isCustomAttributeFunctions[i];
-      if (isCustomAttributeFn(attributeName)) {
+  shouldSetAttribute: function(name, value) {
+    if (DOMProperty.isReservedProp(name)) {
+      return false;
+    }
+
+    if (value === null || DOMProperty.properties.hasOwnProperty(name)) {
+      return true;
+    }
+
+    // Prevent aliases, and badly cased aliases like `class` or `cLASS`
+    // from showing up in the DOM
+    if (DOMProperty.aliases.hasOwnProperty(name.toLowerCase())) {
+      return false;
+    }
+
+    switch (typeof value) {
+      case 'boolean':
+        return isDataOrAriaAttribute.test(name);
```

Comment:
After I wrote the regex match, I did a microbench:
https://jsbench.github.io/#75e098a3100abf2c235ac7b8c3a56d2d

I'm not really sure if this is faster or not. The regex bench has to compile the regex every time. 

I'm also curious if there are issues with a long-lived regex like this.